hathor/nanocontracts/contract_accessor.py

```diff
@@ -0,0 +1,280 @@
+#  Copyright 2025 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Collection, Sequence, final
+
+from hathor.nanocontracts.faux_immutable import FauxImmutable, __set_faux_immutable__
+from hathor.nanocontracts.types import BlueprintId, ContractId, NCAction
+
+if TYPE_CHECKING:
+    from hathor.nanocontracts import Runner
+
+
+@final
+class ContractAccessor(FauxImmutable):
+    """
+    This class represents a "proxy contract instance", or a contract accessor, during a blueprint method execution.
+    Calling custom blueprint methods on this class will forward the call to the actual wrapped blueprint via syscalls.
+    """
+    __slots__ = ('__runner', '__contract_id', '__blueprint_id')
+
+    def __init__(
+        self,
+        *,
+        runner: Runner,
+        contract_id: ContractId,
+        blueprint_id: BlueprintId | Collection[BlueprintId] | None,
+    ) -> None:
+        self.__runner: Runner
+        self.__contract_id: ContractId
+        self.__blueprint_id: BlueprintId | Collection[BlueprintId] | None
```

Comment:
I guess you can simplify the code by changing to `self.__blueprint_id: set[BlueprintId] | None`. So you'd convert the arguments to a set here in the constructor.